### PR TITLE
pkg/oci: Use Singleton for getLocalOciStore

### PR DESCRIPTION
**Pull request by burak-ok**
# pkg/oci: Use Singleton for getLocalOciStore

Concurrent calls to different oci.Store instances leads to race condition on the files that are written to the disk. oci.Store is goroutine safe only against its own instance.

cc @mqasimsarfraz

### CODE REVIEW

**Comment by mauriciovasquezbernal** [comment]
What happens when multiple instances of `ig` are run in parallel? Is there also a race condition in that case?

File: pkg/oci/oci.go [hunk 1]
```diff
@@ -79,6 +80,13 @@ const (
 	localhost = "localhost"
 )
 
+var onceLocalOciStoreFunc = sync.OnceValues(func() (*oci.Store, error) {
```
[the remark this answers is not in the document]
Comment (burak-ok) [reply pkg/oci/oci.go]: Good idea to just use the function name :D

**Review by mqasimsarfraz — APPROVED** [review]
LGTM! 

I tested it and it seem to be working expected, I am fine handling `ig` in seperate PR but we should open and issue for it:


## with this change


```
$ make INTEGRATION_TESTS_PARAMS="-run TestRun -no-deploy-ig -no-deploy-spo" IMAGE_TAG=latest GADGET_TAG=latest integration-tests
...
--- PASS: TestRunSchedCLS (38.37s)
PASS
Cleaning up...
Start command(DeleteRemainingTestNamespace): kubectl delete ns -l scope=ig-integration-tests
Wait for command(DeleteRemainingTestNamespace)
Command returned(DeleteRemainingTestNamespace):

No resources found

ok  	github.com/inspektor-gadget/inspektor-gadget/integration/k8s	38.665s
```

## w/o this change

```
$ make INTEGRATION_TESTS_PARAMS="-run TestRun -no-deploy-ig -no-deploy-spo" IMAGE_TAG=latest GADGET_TAG=latest integration-tests
...
=== NAME  TestRunTraceOOMKill
    teststeps.go:95: [2024-05-06 12:55:30.187113546 +0000 UTC] Stopping "StartRunTraceOOMKillGadget"
    command.go:521: Stop command(StartRunTraceOOMKillGadget)
    command.go:523: Command returned(StartRunTraceOOMKillGadget):
        time="2024-05-06T14:55:06+02:00" level=info msg="Experimental features enabled"
        Error: fetching gadget information: getting gadget info: rpc error: code = Unknown desc = getting gadget info: initializing and preparing operators: instantiating operator "oci": getting manifest: getting index: getting image list descriptor: resolving image "ghcr.io/inspektor-gadget/gadget/trace_oomkill:latest": not found
...
--- PASS: TestRunSchedCLS (35.80s)
FAIL
Cleaning up...
Start command(DeleteRemainingTestNamespace): kubectl delete ns -l scope=ig-integration-tests
Wait for command(DeleteRemainingTestNamespace)
Command returned(DeleteRemainingTestNamespace):

No resources found

FAIL	github.com/inspektor-gadget/inspektor-gadget/integration/k8s	36.125s
FAIL
make: *** [Makefile:278: integration-tests] Error 1
```

**Comment by mqasimsarfraz** [comment]
> > What happens when multiple instances of `ig` are run in parallel? Is there also a race condition in that case?
> 
> Correct. And AFAIK that is not in the scope of `oras-go`. So we would need to implement that with e.g. filesystem locking or lock files

"Multi-process support is also another non-trivial work and it is not on the roadmap of oras-go as it requires an atomicity source such as databases, file systems with O_EXCL support, and etc." - https://github.com/oras-project/oras-go/issues/472#issuecomment-1876251180

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM. Thanks for catching and fixing it folks!

**Pull request merged**